### ISSUE TITLE
[SSR] Wrong documentation (+ possible bug)

### ISSUE THREAD

**Issue by KevinBLT** [solidjs/solid#1797]
### Describe the bug

I was playing around with `renderToStream` and came along this page: https://docs.solidjs.com/references/concepts/ssr/streaming

When using the exact example at the bottom (without `<Suspense>`), it will (as I already wondered before I tried) throw this error:
```
return ssr(_tmpl$, ssrHydrationKey(), escape(data().title));
                                                     ^
TypeError: Cannot read properties of undefined (reading 'title')
```

Which was expected by me, but this is in this documentation.

Another thing, which I didn't know: The `renderToStream()` function will not work, if I only have a resource without `<Suspense>`  around it. Then actually nothing will ever come (at least without hydration). Is that correct behavior?

This will not work:  
```jsx
import { HydrationScript } from "solid-js/web";
import { Suspense, createResource } from 'solid-js';

async function fu() {
  console.log('Yes this will call');

  await new Promise(r => setTimeout(r, 4000));
  
  return 'never seen';
}

export default () => {

  const [data] = createResource(fu);

  return  <html lang="en">
    <head>
      <title>🔥 Solid SSR 🔥</title>
      <meta charset="UTF-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <HydrationScript />
    </head>
    <body>
      <span>{ data() }</span>   
    </body>
  </html>
}
```

Although the fetcher is called, `data()` is never resolved and won't show. If it's in a `<Suspense>` it works, but I would say this should not be a must?

### Your Example Website or App

Code above

### Steps to Reproduce the Bug or Issue

Create SSR with the code above.

### Expected behavior

Stream the value of resolved promise for resource `data()`

### Platform

- OS: macOS
- Chrome
- node --version v20.1.0

**Comment by lxsmnsyc** [comment]
>  If it's in a <Suspense> it works, but I would say this should not be a must?

It is a must. Suspense allows the SSR to know which resources must be awaited.

**Comment by KevinBLT** [comment]
I would have thought, that the resource itself to know this. It offers "deferStream" boolean which will make it work, 
so at some point during execution the resource is "seen" by the stream to say so. And if it's seen (which it must to defer), it could at least prevent the fetch execution to avoid using server resources for nothing. Anyway, it feels a bit odd in a way. 

At least the documentation must be wrong, though.
This part must be wrong then, or am I wrong?
<img width="769" alt="image" src="https://github.com/solidjs/solid/assets/12692409/5222e2b1-c094-4d66-8f24-09e1ddceedc4">

**Comment by ryansolid** [comment]
Yeah the docs are incorrect first in not null checking and unclear about the role of Suspense in this whole process. I think I can safely moving this to the docs-next repo as feedback.
